File: 13_RomanToInteger/13_v1.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        alpha = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000,
        }
        
        buf = 0
        sum = 0
        
        for c in s:
            if buf == 0:
                buf = alpha[c]
            elif buf < alpha[c]:
                sum = sum - buf + alpha[c]
                buf = 0
            else:
                sum = sum + buf
                buf = alpha[c]
            
        sum = sum + buf
        return sum

File: 13_RomanToInteger/13_v1.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    var alpha = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000,
    }
    
    var buf = 0, sum = 0
    
    for (let i = 0; i < s.length; i++){
        var num = alpha[s[i]]
        if (buf === 0){
            buf = num
        } else if (buf < num){
            sum = sum - buf + num
            buf = 0
        } else {
            sum += buf
            buf = num
        }
        console.log(sum)
    }
    
    sum += buf
    return sum
};
